mesonbuild/compilers/mixins/intel.py

```diff
@@ -41,26 +41,32 @@
 # https://software.intel.com/en-us/fortran-compiler-developer-guide-and-reference-g
 # https://software.intel.com/en-us/fortran-compiler-developer-guide-and-reference-traceback
 # https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html
-clike_optimization_args = {
-    '0': ['-O0'],
-    'g': ['-O0'],
-    '1': ['-O1'],
-    '2': ['-O2'],
-    '3': ['-O3'],
-    's': ['-Os'],
-}  # type: typing.Dict[str, typing.List[str]]
 
 
-# Tested on linux for ICC 14.0.3, 15.0.6, 16.0.4, 17.0.1, 19.0.0
 class IntelGnuLikeCompiler(GnuLikeCompiler):
+    """
+    Tested on linux for ICC 14.0.3, 15.0.6, 16.0.4, 17.0.1, 19.0
+    debugoptimized: -g -O2
+    release: -O3
+    minsize: -O2
+    """
 
     BUILD_ARGS = {
         'plain': [],
-        'debug': ["-g", "-O0", "-traceback"],
-        'debugoptimized': ["-g", "-O1", "-traceback"],
-        'release': ["-O2"],
-        'minsize': ["-Os"],
+        'debug': ["-g", "-traceback"],
+        'debugoptimized': ["-g", "-traceback"],
+        'release': [],
+        'minsize': [],
         'custom': [],
+    }  # type: typing.Dict[str, typing.List[str]]
```

Comment:
I'd really like to use this pattern for all of these, where the method is described in `Compiler`, and the subclasses just override the dict if necessary.

Reply:
I think you're saying to make these optimization_args the default? Maybe that would be better for another PR in case of unexpected side effect. 
That is, I do think we should emit `-O0` in compilers.py:clike_optimization_args instead of no option, to be more explicit about intent. Other compilers could also optimize by default, which is not what's intended by Meson.